src/app/clusters/resource-monitoring-server/tests/TestResourceMonitoringCluster.cpp

```diff
@@ -0,0 +1,200 @@
+            TEMPORARY_RETURN_IGNORED item.SetProductIdentifierValue(CharSpan::fromCharString("PRODUCT_4"));
+            break;
+        default:
+            return CHIP_ERROR_PROVIDER_LIST_EXHAUSTED;
+            break;
+        }
+        }
+        mIndex++;
+        return CHIP_NO_ERROR;
+    }
+};
+
+struct TestResourceMonitoringCluster : public ::testing::Test
+{
+    static void SetUpTestSuite() { ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR); }
+
+    static void TearDownTestSuite() { chip::Platform::MemoryShutdown(); }
+
+    void SetUp() override
+    {
+        TestServerClusterContext context;
+        ScopedSafeAttributePersistence persistence(context);
+        ASSERT_EQ(activatedCarbonFilterMonitoring.Startup(testContext.Get()), CHIP_NO_ERROR);
+        activatedCarbonFilterMonitoring.SetReplacementProductListManagerInstance(&replacementProductListManager);
+    }
+
+    void TearDown() override { activatedCarbonFilterMonitoring.Shutdown(); }
+
+    chip::Test::TestServerClusterContext testContext;
+
+    EndpointId kRootEndpointId = 1;
+
+    ResourceMonitoringCluster activatedCarbonFilterMonitoring;
+
+    ImmutableReplacementProductListManager replacementProductListManager;
+
+    TestResourceMonitoringCluster() :
+        activatedCarbonFilterMonitoring(
+            kRootEndpointId, ActivatedCarbonFilterMonitoring::Id,
+            BitFlags<ResourceMonitoring::Feature>{ static_cast<uint32_t>(kResourceMonitoringFeatureMap.to_ulong()) },
+            OptionalAttributeSet(), ResourceMonitoring::DegradationDirectionEnum::kDown, true)
+    {}
+};
+} // namespace
+
+TEST_F(TestResourceMonitoringCluster, AttributeTest)
+{
+    ReadOnlyBufferBuilder<DataModel::AttributeEntry> attributes;
+    ASSERT_EQ(activatedCarbonFilterMonitoring.Attributes(ConcreteClusterPath(kRootEndpointId, ActivatedCarbonFilterMonitoring::Id),
+                                                         attributes),
+              CHIP_NO_ERROR);
+
+    ReadOnlyBufferBuilder<DataModel::AttributeEntry> expected;
+    AttributeListBuilder listBuilder(expected);
+
+    AttributeListBuilder::OptionalAttributeEntry optionalAttributesEntries[] = {
+        { true, Condition::kMetadataEntry },
+        { true, DegradationDirection::kMetadataEntry },
+        { true, ReplacementProductList::kMetadataEntry },
+    };
+
+    ASSERT_EQ(
+        listBuilder.Append(Span(ActivatedCarbonFilterMonitoring::Attributes::kMandatoryMetadata), Span(optionalAttributesEntries)),
+        CHIP_NO_ERROR);
+    ASSERT_TRUE(chip::Testing::EqualAttributeSets(attributes.TakeBuffer(), expected.TakeBuffer()));
+}
+
+TEST_F(TestResourceMonitoringCluster, ReadAttributeTest)
+{
+    ClusterTester tester(activatedCarbonFilterMonitoring);
+
+    uint16_t revision{};
+    ASSERT_EQ(tester.ReadAttribute(Globals::Attributes::ClusterRevision::Id, revision), CHIP_NO_ERROR);
+
+    uint32_t features{};
+    ASSERT_EQ(tester.ReadAttribute(FeatureMap::Id, features), CHIP_NO_ERROR);
```

Comment:
for some of these, could we have an `EXPECT_EQ` on the value? e.g. revision is known, maybe features as well. No idea about the rest.